### DISCUSSION TITLE
Don't understand the document

### DISCUSSION THREAD

**Discussion by webB1an** [ory/hydra#2768] [Q&A] [answered]
Hi！I‘m confused by the document [The OAuth 2.0 Token Endpoint
](https://www.ory.sh/hydra/docs/next/reference/api/#operation/oauth2Token)

> Do not implement a client for this endpoint yourself. Use a library. There are many libraries available for any programming language

What's the meaning of this document？Is it meaning that I can't use this token endpoint or any other?

**Comment by gi-wg2** [comment]
It means that you should see if you could find a nice OAuth 2.0 client for you language instead of making your own.

That is when you want to create the auth url, exchange auth code to token and such, there are plenty of libraries that do this already. Making your own is error prone.

**Reply by webB1an** [reply]
Thank you for your reply!

Can I request the `The OAuth 2.0 Token Endpoint` url when I exchange code to token like this:

```
const GetToken = (param) => {
  return request({
    method: 'post',
    url: '/oauth2/token',
    headers: {
      'Content-Type': 'application/x-www-form-urlencoded'
    },
    data: param
  })
}
```

**Comment by Benehiko** [comment, marked as answer]
Hi @webB1an 

As @gi-wg2 pointed out, you should use an oauth 2.0 library. 

Also see this related discussion https://github.com/ory/hydra/discussions/2740
